src/client/cockpit-beiboot

```diff
@@ -0,0 +1,312 @@
+            self.router.ssh_peer.transport.write(self.payload)
+
+        elif command == 'cockpit.report-exists':
+            file_status, = args
+            # Constructed to throw KeyError if an unexpected file is requested
+            # ...please don't look at this code...
+            Package.CONDITIONS = {
+                'path-exists': lambda filename: file_status[filename],
+                'path-not-exists': lambda filename: not file_status[filename],
+            }
+            self.router.packages = Packages()  # type: ignore[attr-defined]
+            self.router.routing_rules.insert(0, ChannelRoutingRule(self.router, [PackagesChannel]))
+
+
+class SshPeer(Peer):
+    def __init__(self, router: Router, destination: str):
+        self.destination = destination
+        super().__init__(router)
+
+    async def do_connect_transport(self) -> asyncio.Transport:
+        beipack_filename, bridge_beipack_xz = get_bridge_beipack_xz()
+        agent = ferny.InteractionAgent(AuthorizeResponder(self.router, bridge_beipack_xz))
+
+        # We want to run a python interpreter somewhere...
+        cmd: Sequence[str] = ('python3', '-ic', '# cockpit-bridge')
+        env: Sequence[str] = ()
+
+        # Remote host?  Wrap command with SSH
+        if self.destination != 'localhost':
+            ssh_askpass = ensure_ferny_askpass()
+            env = (
+                f'SSH_ASKPASS={ssh_askpass!s}',
+                'DISPLAY=x',
+                'SSH_ASKPASS_REQUIRE=force',
+            )
+            cmd = ('ssh', self.destination, shlex.join(cmd))
+
+        # Running in flatpak?  Wrap command with flatpak-spawn --host
+        if os.path.exists('/.flatpak-info'):
+            cmd = ('flatpak-spawn', '--host',
+                   *(f'--env={kv}' for kv in env),
+                   *cmd)
+            env = ()
+
+        logger.debug("Launching command: cmd=%s env=%s", cmd, env)
+        transport = await self.spawn(cmd, env, stderr=agent, start_new_session=True)
+
+        # Send the first-stage bootloader
+        stage1 = bootloader.make_bootloader([
+            ('report_exists', [list(get_interesting_files())]),
+            ('boot_xz', [beipack_filename, len(bridge_beipack_xz)])
+        ], gadgets=BEIBOOT_GADGETS)
+        transport.write(stage1.encode())
+
+        # Wait for "init" or error, handling auth and beiboot requests
+        await agent.communicate()
+
+        return transport
+
+
+class SshBridge(Router):
+    packages: Packages
+    ssh_peer: SshPeer
+
+    def __init__(self, args: argparse.Namespace):
+        self.ssh_peer = SshPeer(self, args.destination)
+
+        super().__init__([
+            DefaultRoutingRule(self, self.ssh_peer),
+        ])
+
+    def do_send_init(self):
+        pass  # wait for the peer to do it first
+
+    def do_init(self, message):
+        self.ssh_peer.write_control(**message)
+
+
+async def run(args) -> None:
+    logger.debug("Hi. How are you today?")
```

Comment:
debugging leftover?

Reply:
Nope.  An important part of our branding.